isthmus/src/test/resources/extensions/functions_custom.yaml

```diff
@@ -6,19 +6,58 @@ types:
 
 scalar_functions:
   - name: "custom_scalar"
-    description: "a custom scalar functions"
+    description: "a custom scalar function"
     impls:
     - args:
         - name: some_arg
           value: string
       return: string
+  - name: "custom_scalar_any"
+    description: "a custom scalar function that takes any argument input"
+    impls:
+    - args:
+        - name: some_arg
+          value: any1
+      return: string
   - name: "to_b_type"
     description: "converts a nullable a_type to a b_type"
     impls:
       - args:
           - name: arg1
             value: u!a_type?
         return: u!b_type
+  - name: "custom_scalar_listany"
+    description: "custom function that takes list of any"
+    impls:
+      - args:
+          - name: list
+            value: LIST<any1>
+        return: LIST<any1>
+  - name: "custom_scalar_listany_any"
+    description: "custom function that takes list of any and an any scalar"
+    impls:
+      - args:
+          - name: list
+            value: LIST<any1>
+          - name: val
+            value: any1
+        return: LIST<any1>
+  - name: "custom_scalar_liststring"
+    description: "custom function that takes list of string"
+    impls:
+      - args:
+          - name: list
+            value: LIST<string>
+        return: LIST<string>
+  - name: "custom_scalar_liststring_any"
```

Comment:
Trying to make sure the Calcite function definitions matched the signatures was tricky because I had to reference back and forth between the code and the extension. What do you think about encoding the return type with a pattern like `custom_scalar_liststring_any_to_liststring`.

Reply:
+1, much less cognitive load